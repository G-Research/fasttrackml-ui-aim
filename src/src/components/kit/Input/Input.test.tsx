import _ from 'lodash-es';

import { render, fireEvent } from '@testing-library/react';

import InputWrapper from './Input';

// Test number type
describe('<InputWrapper /> -', () => {
  test('Renders Correctly', () => {
    const { asFragment } = render(
      <InputWrapper
        value={5}
        type='number'
        labelAppearance='top-labeled'
        size='small'
        label='test label'
        topLabeledIconName='circle-question'
        labelHelperText='Helper text for tooltip'
        showMessageByTooltip
        onChange={_.noop}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('onChange event works properly', () => {
    const min = 10;
    const max = 100;
    const zero = 0;
    let result: any = null;

    const minValidationPattern = {
      errorCondition: (value: any) => value < 10,
      errorText: `Value should be equal or greater then ${min}`,
    };

    const maxValidationPattern = {
      errorCondition: (value: any) => value > max,
      errorText: `Value should be equal or smaller then ${max}`,
    };

    const zeroValidationPattern = {
      errorCondition: (value: any) => value === zero,
      errorText: `Value can't be equal to ${zero}`,
    };

    const regexpEightValidationPattern = {
      errorCondition: new RegExp(/[8]/g),
      errorText: 'Value should not contain digit 8',
    };

    const { getByTestId } = render(
      <InputWrapper
        type='number'
        labelAppearance='top-labeled'
        size='small'
        label='test label'
        topLabeledIconName='circle-question'
        labelHelperText='Helper text for tooltip'
        showMessageByTooltip
        onChange={(e, value, metadata) => {
          result = metadata;
        }}
        validationPatterns={[
          regexpEightValidationPattern,
          minValidationPattern,
          maxValidationPattern,
          zeroValidationPattern,
        ]}
      />,
    );

    const input = getByTestId('inputWrapper') as HTMLInputElement;

    // Min validation pattern checking
    fireEvent.change(input, { target: { value: 9 } });
    expect(input.value).toBe('9');
    expect(result?.isValid).toBe(false);
    expect(result?.messages[0]?.text).toBe(minValidationPattern.errorText);

    // Max validation pattern checking
    fireEvent.change(input, { target: { value: 101 } });
    expect(result?.isValid).toBe(false);
    expect(result?.messages[0]?.text).toBe(maxValidationPattern.errorText);

    // Zero validation pattern checking
    fireEvent.change(input, { target: { value: 0 } });
    expect(result?.isValid).toBe(false);
    expect(result?.messages[0]?.text).toBe(minValidationPattern.errorText);
    expect(result?.messages[1]?.text).toBe(zeroValidationPattern.errorText);

    // Regexp validation pattern checking
    fireEvent.change(input, { target: { value: 18 } });
    expect(result?.isValid).toBe(false);
    expect(result?.messages[0]?.text).toBe(
      regexpEightValidationPattern.errorText,
    );
  });
});
