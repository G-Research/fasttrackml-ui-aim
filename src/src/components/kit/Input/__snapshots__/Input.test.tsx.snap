// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputWrapper /> - Renders Correctly 1`] = `
<DocumentFragment>
  <div
    class="InputWrapper InputWrapper_topLabeled InputWrapper_small"
  >
    <div
      class="InputWrapper_topLabelCnt"
    >
      <span
        class="Text Text__size_10 Text__weight_400 Text__color_primary_70"
      >
        test label:
      </span>
      <div
        class=""
        title="Helper text for tooltip"
      >
        <i
          class="Icon__container icon-circle-question "
        />
      </div>
    </div>
    <div
      class="InputWrapper_textFieldCnt InputWrapper_textFieldCnt_small"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink="true"
        >
          test label
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="inputWrapper"
            type="number"
            value="5"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
            >
              <span>
                test label
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
